joomla/joomla-cms
# Support collection of custom stats in stats frontend module

#### Summary of Changes

Similar to how the backend module triggers an event to add additional items to its output, this adds support for the `mod_stats` module to do the same.

A new `onGetStats` event is published with a single parameter, a context (hardcoded to `mod_stats_admin`) and any plugin responding to this event is expected to return an array matching this schema:

``` php
array(
    'title' => 'Stats Item Title',
    'data' => 'Data to be displayed',
    'icon' => 'Optional IcoMoon font icon for the row, defaulting to "info" if not set',
);
```

The title and data attributes MUST be defined or a row is not added.

Similar to the quickicons implementation, items can NOT be removed via the plugin event.

#### Testing Instructions

Add the following method to a system plugin:

``` php
    public function onGetStats($context)
    {
        return array(
            array(
                'title' => 'Plays Audio',
                'data'  => JText::_('JNO'),
                'icon'  => 'music',
            ),
        );
    }
```

With the admin stats module enabled, you should see a new row with this data.

Based on the same thing that @mbabker did for the backend: https://github.com/joomla/joomla-cms/pull/10870

The weblinks part is moved here: https://github.com/joomla-extensions/weblinks/pull/333/files to be included in the 3.7 version.
